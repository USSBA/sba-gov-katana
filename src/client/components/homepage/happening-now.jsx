import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Grid, Row, Col } from 'react-bootstrap';
import ModifiedCarousel from "../helpers/carousel.jsx";
import * as ContentActions from "../../actions/content.js";
import styles from "../../styles/homepage/styles.scss";

const contentProperty = "happeningNow";

class HappeningNow extends React.Component {
  constructor() {
    super();
    this.state = {};
  }
  componentDidMount() {
    this.props.actions.fetchContentIfNeeded(contentProperty, "frontpageslides", {});
  }

  render() {
    let items = [];
    let mdSize = 3;
    if (this.props.happeningNow) {
      items = this.props.happeningNow;
      if (items.length === 3) {
        mdSize = 4;
      }
    }

    let carouselItems = items.map(function(item) {
      return {
        title: item.title,
        imageSrc: item.image,
        imageAlt: item.imageAlt,
        href: item.url
      };
    });
    let me = this;
    return (<div className={ styles.happeningNow }>
              <Grid fluid>
                <Row>
                  <Col xs={ 12 } xsHidden smHidden>
                  <p className={ styles.happeningNowTitle }>What's happening now.</p>
                  </Col>
                  <Col xs={ 12 } mdHidden lgHidden>
                    <p className={ styles.happeningNowTitleMobile }>What's happening now.</p>
                  </Col>
                </Row>
                <Row>
                  { items.map(function(item) {
                      return (<Col xsHidden smHidden md={ mdSize }>
                              <a href={ item.url }>
                                <img className="img-responsive" src={ item.image } alt={ item.imageAlt }></img>
                              </a>
                              </Col>);
                    }) }
                </Row>
                <Row>
                  { items.map(function(item) {
                      return <Col xsHidden smHidden md={ mdSize }>
                             <p className={ styles.happeningNowItemTitle }>
                               { item.title }
                             </p>
                             </Col>;
                    }) }
                </Row>
                <Row>
                  { items.map(function(item) {
                      return <Col xsHidden smHidden md={ mdSize }>
                             <a href={ item.url } className={ "btn btn-default " + styles.happeningNowLearnMore }>LEARN MORE</a>
                             </Col>;
                    }) }
                </Row>
                <Row>
                  <Col xs={ 12 } sm={ 12 } mdHidden lgHidden>
                  <div className={ styles.happeningNowMobile }>
                    { items.map(function(item, i) {
                        return (<div className={ styles.happeningNowSection } key={ i }>
                                  <a href={ item.url }>
                                    <img className="img-responsive" src={ item.image } alt={ item.imageAlt }></img>
                                  </a>
                                  <p className={ styles.happeningNowItemTitleMobile }>
                                    { item.title }
                                  </p>
                                  <a href={ item.url } className={ "btn btn-default " + styles.happeningNowLearnMore }>LEARN MORE</a>
                                </div>);
                      }) }
                  </div>
                  </Col>
                </Row>
                {/*<Row>
                  <Col xs={ 12 } sm={ 12 } mdHidden lgHidden>
                  <div className="happeningNowCarousel">
                    <ModifiedCarousel imageStyle={ styles.carouselImage } titleStyle={ styles.happeningNowCarouselItemTitle } items={ carouselItems } />
                  </div>
                  </Col>
                </Row>*/}
              </Grid>
            </div>);
  }
}

function mapReduxStateToProps(reduxState) {
  return {
    happeningNow: reduxState.contentReducer[contentProperty]
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(ContentActions, dispatch)
  };
}
export default connect(mapReduxStateToProps, mapDispatchToProps)(HappeningNow);
