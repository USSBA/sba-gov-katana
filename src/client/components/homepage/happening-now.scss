@import "../../styles/buttons.scss";

.happeningNow {
    margin-top: 5vw;
    padding-left: 0;
    padding-right: 0;
    width: 100%;
}

/* Desktop */
.happeningNowDesktop{
    padding-left: calc(10.4vw - 15px);
    padding-right: calc(10.4vw - 15px);
}

.happeningNowTitle{
    font-size: 5.2vw;
    font-weight: bold;
    padding-left: calc(10.4vw - 15px);
    padding-bottom: 2vw;
}

.happeningNowItemTitle{
    color: black;
    font-size: 1.2vw;
    font-weight: 700;
    padding-top: 1vw;
    padding-bottom: 1vw;
}

.happeningNowLearnMore{
    composes: homeBtnBlue;
    font-weight: 600;
    padding: 10px 20px;
}

/* Mobile */

.happeningNowMobile{
    margin-top: 2vw;
}

.happeningNowTitleMobile{
    font-size: 7.47vw;
    // line-height: 8vw;
    font-weight: bold;
    padding-bottom: 2vw;
    padding-left: 12.53vw;
}
