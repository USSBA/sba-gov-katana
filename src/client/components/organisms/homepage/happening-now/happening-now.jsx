import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import styles from './happening-now.scss'
import * as ContentActions from '../../../../actions/content'
import { BasicLink, Carousel, SmallPrimaryButton } from 'atoms'
import { eventCategories } from '../../../../services/constants'

const contentProperty = 'happeningNow'

class HappeningNow extends React.Component {
  constructor() {
    super()
    this.state = {}
  }
  componentDidMount() {
    this.props.actions.fetchContentIfNeeded(contentProperty, 'frontpageslides')
  }

  makeDesktopImage(item, index, desktopStyle) {
    return (
      <div className={desktopStyle} key={index}>
        <BasicLink url={item.url}>
          <img alt={item.imageAlt} src={item.image} />
        </BasicLink>
      </div>
    )
  }

  makeDeskopTitle(item, index, desktopStyle) {
    return (
      <div className={desktopStyle} key={index}>
        <p className={styles.itemTitleDesktop} key={index}>
          {item.title}
        </p>
      </div>
    )
  }
  makeDesktopButton(item, index, desktopStyle) {
    const buttonText = 'Learn more'.toUpperCase()
    return (
      <div key={index} className={desktopStyle}>
        <SmallPrimaryButton
          eventConfig={{
            category: [eventCategories.frontPage, 'Happening-Now'].join('-'),
            action: `Click: ${buttonText} ${index}`
          }}
          extraClassName={styles.buttonDesktop}
          key={index}
          text={buttonText}
          url={item.url}
        />
      </div>
    )
  }

  render() {
    let items = this.props.happeningNow || []
    let size = items.length

    if (size === 0) {
      return <div />
    }

    let desktopStyle = size === 3 ? styles.itemDesktopThree : styles.itemDesktopFour
    let me = this
    let title = "What's happening now."
    return (
      <div className={styles.container}>
        <div className={styles.containerDesktop}>
          <h2 className={styles.titleDesktop}>{title}</h2>
          <div>
            {items.map((item, index) => {
              return me.makeDesktopImage(item, index, desktopStyle)
            })}
          </div>
          <div key={2}>
            {items.map((item, index) => {
              return me.makeDeskopTitle(item, index, desktopStyle)
            })}
          </div>
          <div key={3}>
            {items.map((item, index) => {
              return me.makeDesktopButton(item, index, desktopStyle)
            })}
          </div>
        </div>
        <div className={styles.containerMobile}>
          <h2 className={styles.titleMobile}>{title}</h2>
          <div className={styles.itemsMobile}>
            <Carousel items={items} />
          </div>
        </div>
      </div>
    )
  }
}

function mapReduxStateToProps(reduxState) {
  return { happeningNow: reduxState.contentReducer[contentProperty] }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(ContentActions, dispatch)
  }
}
export { HappeningNow }
export default connect(mapReduxStateToProps, mapDispatchToProps)(HappeningNow)
