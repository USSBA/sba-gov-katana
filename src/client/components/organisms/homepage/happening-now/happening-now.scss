.container {
  visibility: visible;
}

/* Desktop */
.containerDesktop {
  @include grid-media($sba-grid-small) {
    display: none;
  }
}

.titleDesktop {
  @include grid-column(12, $sba-grid-medium);
  text-align: left;
  font-size: 5.2vw;
  font-weight: 900;
}

.itemTitleDesktop {
  @include font-size($h5-font-sizes);
  color: black;
  font-weight: 700;
  padding-top: $margin-20;
  padding-bottom: $margin-20;
}

.itemDesktopFour {
  @include grid-column(3, $sba-grid-medium);
  @include grid-push(0, $sba-grid-medium);
}

.itemDesktopThree {
  @include grid-column(4, $sba-grid-medium);
  @include grid-push(0, $sba-grid-medium);
}

.buttonDesktop {
}

/* Mobile */
.containerMobile {
  margin-top: 2vw;
  @include grid-media($sba-grid-small) {
    width: 100%;
  }
  @include grid-media($sba-grid-medium) {
    display: none;
  }
  @include grid-media($sba-grid-large) {
    display: none;
  }
}

.titleMobile {
  font-size: 7.47vw;
  font-weight: bold;
  padding-bottom: 2vw;
  padding-left: 12.53vw;
}
