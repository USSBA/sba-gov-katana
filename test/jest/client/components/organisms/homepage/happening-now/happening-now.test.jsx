import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { connect } from 'react-redux'

import { HappeningNow } from '../../../../../../../src/client/components/organisms/homepage/happening-now/happening-now.jsx'

const items = [
  {
    image: 'image.png',
    imageAlt: 'some image',
    title: 'Title',
    url: 'https://sba.gov'
  }
]

const jsx = (
  <HappeningNow
    happeningNow={items}
    actions={{
      fetchContentIfNeeded() {
        return
      }
    }}
  />
)

describe('happening now section', () => {
  test('renders correctly', () => {
    const tree = renderer.create(jsx).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('buttons render correct event config', () => {
    const buttons = shallow(jsx)
      .find('SmallPrimaryButton')
      .map((button, index) => {
        const eventConfig = {
          category: 'Front-Page-Happening-Now',
          action: `Click: LEARN MORE ${index}`
        }
        expect(button.props().eventConfig).toEqual(eventConfig)
      })
  })
})
