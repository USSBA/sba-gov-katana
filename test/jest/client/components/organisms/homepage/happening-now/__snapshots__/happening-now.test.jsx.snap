// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happening now section renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="containerDesktop"
  >
    <h2
      className="titleDesktop"
    >
      What's happening now.
    </h2>
    <div>
      <div
        className="itemDesktopFour"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <img
            alt="some image"
            src="image.png"
          />
        </a>
      </div>
    </div>
    <div>
      <div
        className="itemDesktopFour"
      >
        <p
          className="itemTitleDesktop"
        >
          Title
        </p>
      </div>
    </div>
    <div>
      <div
        className="itemDesktopFour"
      >
        <button
          autoFocus={false}
          className="SmallPrimaryButton buttonDesktop "
          disabled={undefined}
          formTarget="_self"
          id={undefined}
          onKeyPress={[Function]}
          onTouchTap={[Function]}
          tabIndex="0"
        >
          LEARN MORE
        </button>
      </div>
    </div>
  </div>
  <div
    className="containerMobile"
  >
    <h2
      className="titleMobile"
    >
      What's happening now.
    </h2>
    <div
      className="itemsMobile"
    >
      <div
        className="Slider"
      >
        <div
          className="SliderInner"
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="SliderSlides"
            style={
              Object {
                "transform": "translateX(-16.5%)",
                "width": "233.4%",
              }
            }
          >
            <div>
              <a
                onClick={[Function]}
                style={Object {}}
              >
                <img
                  alt="some image"
                  className="SliderItemImage"
                  src="image.png"
                />
              </a>
            </div>
            <div>
              <a
                onClick={[Function]}
                style={Object {}}
              >
                <img
                  alt="some image"
                  className="SliderItemImage"
                  src="image.png"
                />
              </a>
            </div>
            <div>
              <a
                onClick={[Function]}
                style={Object {}}
              >
                <img
                  alt="some image"
                  className="SliderItemImage"
                  src="image.png"
                />
              </a>
            </div>
          </div>
          <a
            className="SliderItemAnchor"
            onClick={[Function]}
            style={Object {}}
          >
            <p
              className="SliderItemTitle"
            >
              Title
            </p>
          </a>
        </div>
        <div
          className="SliderNav"
        >
          <button
            className="SliderNavButton SliderNavButtonActive"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
